src/Catel.Core/Catel.Core.Shared/WeakReferences/WeakEventHandler.cs

```diff
@@ -0,0 +1,173 @@
+﻿namespace Catel
```

Comment:
WeakEventHandler is strongly typed, Usage of Subscribe method is more intuitive to (IDisposable interface suggest correct usage). On the other hand this class is limited to support only 2 types of delegates. In my opinion it should be deleted but some of it pieces should be introduced into WeakEventListener.